recovery/src/main/java/com/kin/ecosystem/recovery/events/BroadcastManagerImpl.java

```diff
@@ -31,46 +29,38 @@
 
 	public BroadcastManagerImpl(@NonNull Activity activity) {
 		this.activity = activity;
+		receiversList = new ArrayList<>();
 	}
 
 	@Override
 	public void register(@NonNull final Listener listener, @ActionName final String actionName) {
-		this.listener = listener;
-		receiver = new BroadcastReceiver() {
+		BroadcastReceiver broadcastReceiver = new BroadcastReceiver() {
 			@Override
 			public void onReceive(Context context, Intent data) {
-				BroadcastManagerImpl.this.listener.onReceive(data);
+				listener.onReceive(data);
 			}
 		};
-		LocalBroadcastManager.getInstance(activity).registerReceiver(receiver, new IntentFilter(actionName));
+		receiversList.add(broadcastReceiver);
+		LocalBroadcastManager.getInstance(activity).registerReceiver(broadcastReceiver, new IntentFilter(actionName));
 	}
 
 	@Override
-	public void unregister() {
-		if (receiver != null && activity != null) {
-			LocalBroadcastManager.getInstance(activity).unregisterReceiver(receiver);
+	public void unregisterAll() {
```

Comment:
should we add option to delete a listener, and not all of them ?

Reply:
as it's internal mechanism let's add it by demand in the future

Reply:
👍